core/src/main/kotlin/org/evomaster/core/problem/rest/service/SecurityRest.kt

```diff
@@ -861,7 +872,159 @@ class SecurityRest {
+                var copy = sliced.copy() as RestIndividual
+                val actionCopy = copy.seeMainExecutableActions().last() as RestCallAction
+
+                val genes = GeneUtils.getAllStringFields(actionCopy.parameters)
+                    .filter { it.staticCheckIfImpactPhenotype() }
+
+                if(genes.isEmpty()){
+                    continue
+                }
+
+                val anySuccess = genes.map { gene ->
+                    gene.setFromStringValue(payload)
+                }.any { it }
+
+                if(!anySuccess){
+                    continue
+                }
+
+                // Try to add a linked GET operation for stored XSS detection
+                //TODO to properly handle POST, we need first to finish the work on CallGraphService
+                if(action.verb == HttpVerb.POST || action.verb == HttpVerb.PUT || action.verb == HttpVerb.PATCH){
+                    copy = tryAttachLinkedGetForStoredXSS(
+                        ind = copy,
+                        path = actionCopy.path,
+                        payload = payload
+                    ) ?: copy
+                }
+
+                copy.modifySampleType(SampleType.SECURITY)
+                copy.ensureFlattenedStructure()
+
+                try {
+                    val evaluatedIndividual = fitness.computeWholeAchievedCoverageForPostProcessing(copy)
+
+                    if (evaluatedIndividual == null) {
+                        log.warn("Failed to evaluate constructed individual in handleStackTraceCheck")
+                        continue@mainloop
+                    }
+
+                    val faultsCategories = DetectedFaultUtils.getDetectedFaultCategories(evaluatedIndividual)
+
+                    if(DefinedFaultCategory.XSS in faultsCategories){
+                        archive.addIfNeeded(evaluatedIndividual)
+                        continue@mainloop
+                    }
+                } catch(e: Exception){
+                    // For certain malformed or encoded requests (such as /api/cleanup/items/<img src=x onerror=alert('XSS')>
+                    // which becomes /api/cleanup/items/%3Cimg%20src=x%20onerror=alert('XSS')%3E), the framework may throw
+                    // a "URI is not absolute" error due to invalid or non-absolute path handling. Such cases can occur when
+                    // XSS-like payloads are injected or when user input is not properly sanitized. The try-catch block ensures
+                    // that these exceptions do not interrupt the main processing loop.
+
+                    log.warn("Failed to evaluate constructed individual in handleStackTraceCheck")
+                }
+            }
+        }
+    }
+
+    private fun tryAttachLinkedGetForStoredXSS(
+        ind: RestIndividual,
+        path: RestPath,
+        payload: String
+    ): RestIndividual? {
+
+        val getIndividuals = RestIndividualSelectorUtils.findIndividuals(
```

Comment:
add `//TODO to properly handle POST, we need first to finish the work on CallGraphService`